packages/frontend-core/src/api/self.js

```diff
@@ -0,0 +1,54 @@
+export const buildSelfEndpoints = API => ({
+  /**
+   * Using the logged in user, this will generate a new API key,
+   * assuming the user is a builder.
+   * @return {Promise<object>} returns the API response, including an API key.
+   */
+  generateAPIKey: async () => {
+    const response = await API.post({
+      url: "/api/global/self/api_key",
+    })
+    return response?.apiKey
+  },
+
+  /**
+   * retrieves the API key for the logged in user.
+   * @return {Promise<object>} An object containing the user developer information.
+   */
+  fetchDeveloperInfo: async () => {
```

Comment:
Should it be just `fetchAPIKey` to match the above / the endpoint?

Reply:
Wanted to have a single endpoint to generate, but a fetch that could fetch the whole doc of developer info. In the future if there is any other developer info can get it all from a single endpoint, but we might have multiple endpoints to re-generate components of it.